Add more unit tests for trainer, fix __init__.py and rename train.py to avoid import conflict

yangaws opened the pull request:
*Issue #, if available:*

*Description of changes:*
1, Add more unit tests for trainer
2, Fix __all__
3, Rename train.py to avoid import names conflict

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mvsusp commented on src/tf_container/__init__.py:
```diff
@@ -11,8 +11,8 @@
 #  permissions and limitations under the License.
 
-from tf_container.train import train
+from tf_container.train_entry_point import train
```
Where is train_entry_point coming from?

winstonaws replied:
This change renames train.py to train_entry_point.py, since otherwise there is no way to access the module tf_container.train. __init__.py imports the train function, which overrides the train module.

This affects our ability to test other functions from train_entry_point.py.

mvsusp commented on src/tf_container/__init__.py:
```diff
@@ -11,8 +11,8 @@
 import tf_container.serve as serve
 
-__all__ = [train, transformer, serve, load_dependencies]
+__all__ = ['train', 'transformer', 'serve', 'load_dependencies']
```
Why is this change necessary?

yangaws replied:
__all__ should include list of names instead of values.

From python doc(https://docs.python.org/2/tutorial/modules.html):

'The import statement uses the following convention: if a package’s __init__.py code defines a list named __all__, it is taken to be the list of module names that should be imported when from package import * is encountered. It is up to the package author to keep this list up-to-date when a new version of the package is released. Package authors may also decide not to support it, if they don’t see a use for importing * from their package. For example, the file sound/effects/__init__.py could contain the following code:

__all__ = ["echo", "surround", "reverse"]'

winstonaws commented on test/unit/test_trainer.py:
```diff
@@ -67,20 +74,59 @@ def trainer(trainer_module):
                                  customer_params=HYPERPARAMETERS.copy())
 
 
+def test_test_trainer_params_passing(trainer_module):
```
test_test_?

yangaws replied:
Fixed. Thanks

yangaws commented:
@winstonaws 

New updates:
1, Added unit test for _resolve_value_for_training_input_fn_parameter functionality
2, I think we already have mock tests for _configure_s3_file_system(). I just updated a little bit to check the order.
